### PR TITLE
Update how to document changes in contribution guide & PR template

### PR DESCRIPTION
**Pull request by lagru**
## Description

In the past I noticed some confusion around how to use [changelist](https://github.com/scientific-python/changelist) and the `release-notes` block in our PRs. This updates the outdated section in our contribution guide and gives a few helpful pointers.

I am open to discuss if this is the best way to address this. Alternatively, we might update our core developer guide to document our usage in more detail and just link to that. 

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
Update description of how to document pull requests
for inclusion in the release notes.
```

### CODE REVIEW

File: CONTRIBUTING.rst
Comment (jarrodmillman) [inline CONTRIBUTING.rst:126]: This seems like overkill to me. I would prefer just saying that if you want to add a longer description of the PR you may.

Comment (mkcor) [reply CONTRIBUTING.rst:126]: I agree with @jarrodmillman. First, API changes typically imply deprecation cycles, which brings us back to the previous sentence/instruction. Then, I'm afraid that whatever "impacts users directly (e.g. bug fixes, performance improvements) covers too broad an area.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mkcor) [inline .github/PULL_REQUEST_TEMPLATE.md:22]:
```suggestion
list each pull request in the release notes. Please refer to
```

File: CONTRIBUTING.rst
Comment (mkcor) [inline CONTRIBUTING.rst:139]:
```suggestion
   By default, `changelist <https://github.com/scientific-python/changelist>`_ uses the
   PR title to summarize your change in the release notes.
   
   If you would like to add a longer description of your change in the release notes,
   you may do so inside a dedicated code block::

       ```release-note
       Write a longer description of the change here.
       ```

   When referencing a public function, class, or parameter, please use the full import
   path, e.g., ``skimage.submodule.function``. Refer to :doc:`/release_notes/index` for
   examples.
```

**Review by mkcor — COMMENTED** [review]
Thanks for improving these instructions, @lagru!

File: CONTRIBUTING.rst
Comment (lagru) [reply CONTRIBUTING.rst:126]: I included

> For trivial changes that are adequately summarized by the PR title, this
   block can be left empty (``...``).

below to make it clear that this is only required if the PR title isn't descriptive enough. Maybe I could have worded / ordered this better.

Basically, I want users to be able to read the New Features, API Changes, Bug fixes, and Performance sections without having to go to each PR to understand if this is relevant to them. E.g. consider https://github.com/scikit-image/scikit-image/pull/7166: out of context, the title

> Fix deprecation for phase_cross_correlation

is vague to users. But the extended summary makes it clear to users what's going on:

> Change number of returned arguments in `skimage.registration.phase_cross_correlation`. The function now always returns the 3 arguments `shift`, `error`, and `phasediff`.

I'll try to come with a wording that makes this clearer and has some tolerance build in.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (lagru) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: "list" directly after "changelist" seems somewhat redundant.

What do you think about

> We use changelist to describe changes in each pull request in our release notes.

File: CONTRIBUTING.rst
Comment (lagru) [reply CONTRIBUTING.rst:139]: Waiting for your thoughts on https://github.com/scikit-image/scikit-image/pull/7267/files#r1429156737 until addressing this.

Comment (mkcor) [reply CONTRIBUTING.rst:139]: Sure. Sorry for submitting a large *block* of suggestions; my intention was originally to *describe* my ideas in full sentences and bullet points, but then I thought that writing them out might be clear enough.

File: CONTRIBUTING.rst
Comment (mkcor) [reply CONTRIBUTING.rst:126]: > below to make it clear that this is only required if the PR title isn't descriptive enough. Maybe I could have worded / ordered this better.

Ok, I got your intention right and I mostly agree. My reasoning is based on putting myself in a new contributor's shoes:

* repeating my support for @jarrodmillman's comment, I don't want to convey that something is (or isn't) *required*: We maintainers can complement whatever needs be complemented when we merge the PR;
* I would avoid calling any PR 'trivial:' In some cases it clearly is, by all standards, but in other, more subtle cases, it may sound demeaning.

Comment (mkcor) [reply CONTRIBUTING.rst:126]: > is vague to users. But the extended summary makes it clear to users what's going on:

Sure, but I would simply *encourage* contributors to add a longer description of their PR...

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mkcor) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: Maybe it's just me and 'compile' is actually fine: I tend to think of the computing meaning *first* but really that's the *second* meaning of 'compile!'

Otherwise:

> We use changelist to log each pull request and its description in the release notes.

I find '*our* release notes' overly possessive :sweat_smile:

**Review by stefanv — APPROVED** [review]
Once feedback is addressed, happy for this to go in.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mkcor) [inline .github/PULL_REQUEST_TEMPLATE.md:23]:
```suggestion
[instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) and past [release notes](https://scikit-image.org/docs/stable/release_notes/index.html) for guidance and examples.
```
(they are "instructions [on how to document changes]")

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mkcor) [inline .github/PULL_REQUEST_TEMPLATE.md:22]:
```suggestion
compile each pull request into an item of the release notes. Please refer to
```

File: CONTRIBUTING.rst
Comment (mkcor) [reply CONTRIBUTING.rst:139]: Ok, I've just re-reviewed this PR and I stand by this suggestion; I have updated/resolved my other comments. Thanks!

File: CONTRIBUTING.rst
Comment (mkcor) [inline CONTRIBUTING.rst:130]:
```suggestion
   changes, we encourage you to describe them in more detail using the
   `release-note` code block within the pull request description; e.g.::
```

**Review by mkcor — APPROVED** [review]
@lagru re-approving just formally! I like your latest changes, I've only left one last suggestion.

**Comment by lagru** [comment]
Great, I'll merge this if there are no additional comments by Tuesday.

**Comment by lagru** [comment]
Thanks everyone.

**Pull request merged**
